### PR TITLE
[jobs] autodown managed job clusters

### PR DESCRIPTION
**Pull request by cg505**
If all goes correctly, the managed job controller should tear down a managed job cluster once the managed job completes. However, if the controller fails somehow (e.g. crashes, is terminated, etc), we don't want to leak resources.

As a failsafe, set autodown on the job cluster. This is not foolproof, since the skylet on the cluster can also crash, but it's likely to catch many cases.

Tested (run the relevant ones):

- [x] Code formatting: `bash format.sh`
- [ ] Any manual or new tests for this PR (please specify below)
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `conda deactivate; bash -i tests/backward_compatibility_tests.sh`

### CODE REVIEW

File: sky/jobs/recovery_strategy.py [hunk 1]
```diff
@@ -302,11 +302,17 @@ def _launch(self,
...
-                           detach_setup=True,
-                           detach_run=True,
-                           _is_launched_by_jobs_controller=True)
+                sky.launch(
+                    self.dag,
+                    cluster_name=self.cluster_name,
+                    # We expect to tear down the cluster as soon as the job is
+                    # finished. However, in case the controller dies, set
+                    # autodown to try and avoid a resource leak.
+                    idle_minutes_to_autostop=5,
```
Comment (Michaelvll) [inline sky/jobs/recovery_strategy.py:311]: One thing we should double-check is the gap between the controller's monitoring/refreshing logic, and this idle minutes should be significantly longer than the refreshing gap. As a safety net, this autodown idle minutes might be good to set to longer, such as 10mins?

Comment (cg505) [reply sky/jobs/recovery_strategy.py]: Controller checks every 20s. Added a comment about this to the PR.

File: sky/jobs/recovery_strategy.py [hunk 2]
```diff
@@ -36,6 +36,11 @@
 # 10 * JOB_STARTED_STATUS_CHECK_GAP_SECONDS = 10 * 5 = 50 seconds
 MAX_JOB_CHECKING_RETRY = 10
 
+# Minutes to job cluster autodown. This should be significantly larger than
+# managed_job_utils.JOB_STATUS_CHECK_GAP_SECONDS, to avoid tearing down the
+# cluster before its status can be updated by the job controller.
+AUTODOWN_MINUTES = 5
```
Comment (Michaelvll) [inline sky/jobs/recovery_strategy.py:42]: For variable private to the module, prepend a `_`.

**Review by Michaelvll — APPROVED** [review]
LGTM! Thanks @cg505 !

**Pull request merged**